helix-term/build.rs

```diff
@@ -1,9 +1,143 @@
 use helix_loader::grammar::{build_grammars, fetch_grammars};
 
+// #[cfg(windows)]
+use std::io::prelude::Write;
+use std::path::Path;
+use std::{io, path::PathBuf, process};
+
 fn main() {
     if std::env::var("HELIX_DISABLE_AUTO_GRAMMAR_BUILD").is_err() {
         fetch_grammars().expect("Failed to fetch tree-sitter grammars");
         build_grammars(Some(std::env::var("TARGET").unwrap()))
             .expect("Failed to compile tree-sitter grammars");
     }
+
+    if cfg!(target_os = "windows") {
+        let rc_exe = find_rc_exe().expect("Windows SDK is to be installed along with MSVC");
+
+        let output =
+            std::env::var("OUT_DIR").expect("Env var OUT_DIR should have been set by compiler");
+        let output_dir = PathBuf::from(output);
+
+        let rc_path = output_dir.join("resource.rc");
+        let icon_path = PathBuf::from("../contrib/helix-256p.ico");
+        write_resource_file(&rc_path, icon_path.to_str().unwrap()).unwrap();
+
+        let resource_file = PathBuf::from(&output_dir).join("resource.lib");
+        compile_with_toolkit_msvc(rc_exe, resource_file, rc_path);
+
+        println!("cargo:rustc-link-search=native={}", output_dir.display());
+        println!("cargo:rustc-link-lib=dylib=resource");
+    }
```

Comment:
this should be moved to a sepreate cfg(windows) windows function (then you can define the function on cfg(not(windows)) to do nothing)

Reply:
Would it be better off like this.. in a separate module with a single entry point and imports also conditionally imported?

Reply:
Thanks!